torch_np/_ndarray.py

```diff
@@ -95,19 +133,38 @@ def __float__(self):
         return float(self._tensor)
 
 
+    ### sequence ###
+    def __len__(self):
+        return self._tensor.shape[0]
+
     ### arithmetic ###
 
     def __add__(self, other):
-        return self._tensor.__add__(other)
+        other_tensor = asarray(other).get()
+        return asarray(self._tensor.__add__(other_tensor))
 
     def __iadd__(self, other):
-        return self._tensor.__add__(other)
+        other_tensor = asarray(other).get()
+        return asarray(self._tensor.__add__(other_tensor))
```

Comment:
```suggestion
        return asarray(self._tensor.__iadd__(other_tensor))
```

Reply:
Probably it'd be best to implement all these binary dunder methods generically to avoid these slips. There's a similar slip in `__rmul__`.

There's also many others in PyTorch, like `__radd__` `__rdiv__` and so on. I reckon all these also exist in NumPy.